### ISSUE TITLE
my app crashed with a realm exception: Top ref not aligned

### ISSUE THREAD

**Issue by makecloud** [realm/realm-java#7608]
### How frequently does the bug occur?

Sometimes

### Description

amazing, amazing

### Stacktrace & log output

```shell
Unable to open a realm at path '/data/data/com.oohlink.player/files/oohlink.realm': Top ref not aligned (179029). top_ref[0]: 774E010155FFAA00, top_ref[1]: 2BB55, mnemonic: 54 2D 44 42, fmt[0]: 9, fmt[1]: 9, flags: 1 Path:Exception backtrace:
<backtrace not supported on this platform>. (Top ref not aligned (179029). top_ref[0]: 774E010155FFAA00, top_ref[1]: 2BB55, mnemonic: 54 2D 44 42, fmt[0]: 9, fmt[1]: 9, flags: 1 Path: /data/data/com.oohlink.player/files/oohlink.realm
Exception backtrace:
<backtrace not supported on this platform>) (/data/data/com.oohlink.player/files/oohlink.realm) in /Users/cm/Realm/realm-java-release/realm/realm-library/src/main/cpp/io_realm_internal_OsSharedRealm.cpp line 101 Kind: ACCESS_ERROR.
        at io.realm.internal.OsSharedRealm.nativeGetSharedRealm(Native Method)
        at io.realm.internal.OsSharedRealm.<init>(SourceFile:9)
        at io.realm.internal.OsSharedRealm.getInstance(SourceFile:5)
        at io.realm.BaseRealm.<init>(SourceFile:18)
        at io.realm.BaseRealm.<init>(SourceFile:1)
        at io.realm.Realm.<init>(SourceFile:1)
        at io.realm.Realm.createInstance(SourceFile:1)
        at io.realm.RealmCache.doCreateRealmOrGetFromCache(SourceFile:23)
        at io.realm.RealmCache.createRealmOrGetFromCache(SourceFile:2)
        at io.realm.Realm.getDefaultInstance(SourceFile:5)
        at com.oohlink.player.sdk.a.b.a.d(SourceFile:9)
        at com.oohlink.player.sdk.a.a.e(SourceFile:3)
        at com.oohlink.player.sdk.manager.i.e(SourceFile:1)
        at com.oohlink.player.sdk.manager.i.a(SourceFile:1)
        at com.oohlink.player.sdk.manager.i$a.a(SourceFile:1)
        at com.oohlink.player.sdk.manager.i$a.c(SourceFile:1)
        at e.a.v.h.c.a(SourceFile:7)
        at e.a.v.e.a.g$a.run(SourceFile:3)
        at e.a.v.g.h.run(SourceFile:2)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:428)
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:278)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:273)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:761)
```

### Can you reproduce the bug?

Not yet

### Version

6.0.2

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

android 7

### Build environment

Android Studio version: ...
Android Build Tools version: ...
Gradle version: ...

**Comment by clementetb** [comment]
We can see that you are using Realm version `6.0.2`, would it be possible to upgrade to a newest version?

**Comment by makecloud** [comment]
> 

you mean the exception, hard to find reason?

**Comment by makecloud** [comment]
upgrade version, cause i upgrade a lot of app that are running online env.  it will make me crazy, cry

**Comment by clementetb** [comment]
There have been many bug fixes since version `6.0.2` and the issue you reported probably has been already fixed. Please try to reproduce the issue with the latest Realm version.
